### ISSUE TITLE
how can i get the f1 score in the paper

### ISSUE THREAD

**Issue by HelloNicoo** [allenai/drug-combo-extraction#47]
I run the code and got only about 20 F1-score

**Comment by viswavi** [comment]
Hi, can you please share the commands you ran (and share the version of the model you used) so I can reproduce this number? After that, I will be able to help you.

**Comment by HelloNicoo** [comment]
`python3 scripts/train.py \
    --model-name pubmedbert_2021 \
    --pretrained-lm microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext \
    --num-train-epochs 20 \
    --lr 2e-4 \
    --batch-size 8 \
    --training-file data/final_train_set.jsonl \
    --test-file data/final_test_set.jsonl \
    --context-window-size 400 \
    --max-seq-length 512 \
    --label2idx data/label2idx.json \
    --seed 2022 \
    --unfreezing-strategy final-bert-layer`
I ran the code with the command above and got this result：
DATALOADER:0 TEST RESULTS
{'accuracy': 0.9745259881019592,
 'f1': 0.34113287925720215,
 'precision': 0.3349769711494446,
 'recall': 0.4231139123439789,
 'test_loss': 0.2403484284877777}
----------------------------------------

**Comment by HelloNicoo** [comment]
I used your command above to run the code and got this result：
![image](https://user-images.githubusercontent.com/42365353/189468181-5396fce8-fa81-4c5f-a0e1-4e2c488a5ddc.png)

**Comment by viswavi** [comment]
Did you run the final two command I sent you above? Those are the ones that give you the final evaluation results

**Comment by viswavi** [comment]
```
(cd scripts && ./eval.sh ../data/final_test_set.jsonl ../checkpoints_pubmedbert_2021/outputs/predictions.jsonl > ../checkpoints_pubmedbert_2021/outputs/eval_partial.txt)
(cd scripts && ./eval.sh ../data/final_test_set.jsonl ../checkpoints_pubmedbert_2021/outputs/predictions.jsonl --exact-match > ../checkpoints_pubmedbert_2021/outputs/eval_exact.txt)
```

**Comment by HelloNicoo** [comment]
Thank you! I got a good f1 score, and I would like to ask where the greedy algorithm mentioned in the paper is embodied. I read the code, but I did not find the specific implementation of this algorithm
![image](https://user-images.githubusercontent.com/42365353/189616002-597fe309-d119-4129-b565-33ee63388b2e.png)

**Comment by HelloNicoo** [comment]
Thank you! I got a good f1-score!

**Comment by aryehgigi** [comment]
Hi @HelloNicoo!
Happy to see you are checking out our work, please notice our [Leaderboard](https://leaderboard.allenai.org/drug_combo/submissions/public) if you wish to participate :)

**Comment by JingqiKang** [comment]
I saw this problem when running the test command
![image](https://user-images.githubusercontent.com/88081081/192706706-17718956-6308-49bc-a72c-6432e5163a91.png)

**Comment by viswavi** [comment]
Hi @JingqiKang, I believe this is a bug. I am discussing with my collaborator @aryehgigi to figure out how to proceed, we should have a fix for you by tomorrow.

**Comment by aryehgigi** [comment]
hey @JingqiKang  i added a fix to allow you to choose an output path (or a default otherwise)
lmk if you encounter any other problem (feel free to reopen this issue or open a new one)
thanks
